src/test/ui/issues/issue-57362-2.stderr

```diff
@@ -0,0 +1,11 @@
+error: implementation of `X` is not general enough
+  --> $DIR/issue-57362-2.rs:22:13
+   |
+LL |     let x = <fn (&())>::make_g(); //~ ERROR not general enough
+   |             ^^^^^^^^^^^^^^^^^^
+   |
+   = note: `X` would have to be implemented for the type `for<'r> fn(&'r ())`
+   = note: but `X` is actually implemented for the type `fn(&'0 ())`, for the specific lifetime `'0`
```

Comment:
I think I would vaguely prefer "for some specific lifetime" when referring to a "numeric" lifetime, but otherwise 💯